### PR TITLE
quantum-espresso add CMake and libxc variants

### LINKED ISSUE #24178: add support for libxc in quantum espresso package
recent versions of quantum espresso support linking with libxc.  spack's qe package currently doesn't appear to support it. 

note : linking with libxc@5: requires patching the source atm, see 2.5.1.1 https://www.quantum-espresso.org/Doc/user_guide/node12.html

/

### PR DESCRIPTION
**Pull request by ye-luo**
Add CMake support for >= v6.8
Add libxc support, closes #24178

As @naromero77 requested in #27607, put myself as the maintainer for now.

### CODE REVIEW

**Review by naromero77intel — APPROVED** [review]

**Review by naromero77 — APPROVED** [review]

**Review by bernhardkaindl — APPROVED** [review]
Approving as this has the approval of the current maintainer.

**Pull request merged**
